CMakeLists.txt

```diff
@@ -1,18 +1,28 @@
-cmake_minimum_required(VERSION 3.16)
-# https://semver.org/
+cmake_minimum_required(VERSION 3.24)
+
 project(farm_ng_core VERSION 0.2.0)
```

Comment:
Maybe we bump this to indicate all these changes? Some projects that build against this package would probably not be compatible out of the box anymore.

Reply:
Happy to do that. I'm not sure if this package has any downstream consumers that are consuming it as a package though.